src/Hit/Git/Branch.hs

```diff
@@ -29,12 +33,53 @@ import qualified Data.Text as T
 -- | @hit new@ command.
 runNew :: Bool -> Text -> IO ()
 runNew isIssue issueOrName = do
+    branchName <- mkBranchName isIssue issueOrName
+    "git" ["checkout", "-b", branchName]
+
+{- | @hit rename@ command.
+
+Renames the current (non-master) branch to the given name or issue with the
+username prefix.
+
+If the current branch is master, it just creates a new branch as in 'runNew'.
+-}
+runRename :: Text -> IO ()
+runRename issueOrName = do
+    curBranch <- getCurrentBranch
+    if curBranch == "master"
+    then runNew False issueOrName
+    else do
+        newBranch <- mkBranchName False issueOrName
+        -- rename a current branch locally
+        "git" ["branch", "-m", curBranch, newBranch]
+        -- check if the old branch is on remote
+        isRemote <- (True <$ "git" ["ls-remote", "--exit-code", "--heads", "origin", curBranch]) $? pure False
```

Comment:
I've created an issue to implement a helper function for this:

https://github.com/kowainik/shellmet/issues/20

Looks like something we use multiple times 🙂